### PR TITLE
Issue/2026 include readme md in module docs

### LINKED ISSUE #72: sphinxcontrib is declared as an old-style namespace package
This uses the deprecated `pkg_resources`. This should be updated. Likely it suffices to delete the `__init__.py` file.

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

Part of https://github.com/inmanta/irt/issues/2026

closes https://github.com/inmanta/inmanta-sphinx/issues/72 closes https://github.com/inmanta/irt/issues/2026

Current PR:
  - Add new command `generate-api-doc-v2` that detects presence of a `README.md` file + generates relevant file structure
  - Relevant changes are in `sphinxcontrib/inmanta/api.py` (other files changes due to running `black`)

Core PR https://github.com/inmanta/inmanta-core/pull/7861
  - Add the patch to include `README.md`
  - Bump `inmanta-sphinx` version

IRT PR https://github.com/inmanta/irt/pull/2113
  - Use new command `generate-api-doc-v2`

Tested locally with:
```
mkvirtualenv docs -p 3.11
git clone git@code.inmanta.com:solutions/modules/vcenter.git
git clone git@github.com:inmanta/inmanta-core.git --branch issue/2026-include-readme-md-in-module-docs
git clone git@github.com:inmanta/inmanta-sphinx.git --branch issue/2026-include-readme-md-in-module-docs
pip install inmanta-module-std -r inmanta-core/requirements.dev.txt  -e inmanta-core -e inmanta-sphinx

mkdir inmanta-core/docs/reference/modules
python -m sphinxcontrib.inmanta.api generate-api-doc-v2 --module_repo $(pwd) --module-name vcenter  --out-dir inmanta-core/docs/reference/modules/

cd inmanta-core/docs
make html
```

Note to self, merge this first then core and irt (they require >= 1.11.0) 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: CHANGELOG.md
Comment (arnaudsjs) [inline CHANGELOG.md]: Why call it `generate-api-doc-v2`? It can be confusing, because api doc can be interpreted at the documentation for our (rest) API. Wouldn't it be better to name if `generate-module-doc-v2`?

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]:
```suggestion
        :param impl: The implementation being documented
```

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py:689]: We should not call this "api doc" because it's a lot more than just the API. For example this might also include the changelog, etc.

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: The output file is not in-line with the docstring. According to the docstring this should be `<module_name>.rst`. Did you test this scenario?

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: Our module tool assumes the file to be named `CHANGELOG.md`.
```suggestion
        os.path.join(module_dir, "CHANGELOG.md"),
```

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: This list also contains directories.
```suggestion
    to_copy = [
```

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: The part about the v2 modules is not true as far as I can see. No files are ever picked up from the Python environment. Did you test this on a v2 module? Is this even going to work? As far as I know we don't include the docs directory, changelog, etc. in the Python packages.

File: ci/Jenkinsfile-test
Comment (arnaudsjs) [inline ci/Jenkinsfile-test:1]: It feels strange to me to keep the old code, but not test it on the CI. Isn't this also a breaking change? I mean, is the old command to build the docs still going to work after this is merged?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: sphinxcontrib/inmanta/api.py
Comment (Hugo-Inmanta) [reply sphinxcontrib/inmanta/api.py]: As discussed in https://inmanta.slack.com/archives/CKRF0C8R3/p1721379733202019?thread_ts=1721379401.623199&cid=CKRF0C8R3 -> assume the source is available for both v1 and v2.

File: ci/Jenkinsfile-test
Comment (Hugo-Inmanta) [reply ci/Jenkinsfile-test:1]: Instead of adding a new command, replaced the old one altogether

File: sphinxcontrib/inmanta/api.py
Comment (Hugo-Inmanta) [reply sphinxcontrib/inmanta/api.py]: Nice catch, tested all 3 scenarios now:
- No readme file
- readme file with autodoc.rst
- readme file without autodoc.rst

File: sphinxcontrib/__init__.py
Comment (sanderr) [inline sphinxcontrib/__init__.py:19]: FYI: https://github.com/inmanta/inmanta-sphinx/issues/72

File: sphinxcontrib/inmanta/api.py
Comment (sanderr) [inline sphinxcontrib/inmanta/api.py]: Most of these should be imported from `collections.abc` now. I especially think that we should import new imports from there.

Comment (sanderr) [reply sphinxcontrib/inmanta/api.py]: Also, `Any` doesn't seem to be used?

File: sphinxcontrib/inmanta/api.py
Comment (sanderr) [inline sphinxcontrib/inmanta/api.py]:
```suggestion
def parse_docstring(docstring: str) -> dict[str, Union[dict[str, str], list[str]]]:
```
Is this what you meant here?

File: sphinxcontrib/inmanta/api.py
Comment (sanderr) [inline sphinxcontrib/inmanta/api.py]: I'm not sure if we even need to open the file. This feels weird to me. I would expect we'd either pass the handle, or leave it to `toml` to read the file.

File: sphinxcontrib/inmanta/api.py
Comment (sanderr) [inline sphinxcontrib/inmanta/api.py:678]: I wonder if we should add an option for the old behavior, in case users already have a `README.md` but are not ready to go to the new flow yet? e.g. `--ignore-readme` or `--autodoc-only`.

**Review by sanderr — CHANGES_REQUESTED** [review]
Looks mostly good to me, some small comments.

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py:666]: This is a breaking change so we need to do a major version bump.

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: This wording gives the impression that each module has to be present as a v1 and a v2 module.
```suggestion
        "The path to the local directory where all module sources (v1 or v2) are stored."
```

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]:
```suggestion
    If the module doesn't contain a README.md or if the `--autodoc-only` option is provided, then the api documentation for this module is generated and placed in a
```

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [inline sphinxcontrib/inmanta/api.py]: Why `dirs_exist_ok=True`?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: sphinxcontrib/inmanta/api.py
Comment (arnaudsjs) [reply sphinxcontrib/inmanta/api.py:666]: Ah. Wait. I think the version bumps of development tools is done at release time and not at development time. You can ignore this comment.

File: sphinxcontrib/inmanta/api.py
Comment (Hugo-Inmanta) [reply sphinxcontrib/inmanta/api.py]: No good reason, removing it

File: sphinxcontrib/inmanta/api.py
Comment (Hugo-Inmanta) [reply sphinxcontrib/inmanta/api.py:678]: Added the option and tested it manually

**Review by arnaudsjs — APPROVED** [review]

File: setup.py
Comment (sanderr) [inline setup.py]: Looking at the [example namespace package](https://github.com/pypa/sample-namespace-packages/blob/master/native/pkg_b/setup.py), I *think* this could be simply
```suggestion
    packages=['sphinxcontrib.inmanta'],
```

**Review by sanderr — CHANGES_REQUESTED** [review]
I'm a bit hesitant on the package restructure. Is it really necessary to move everything into `src`? I wouldn't think so, especially as the example namespace package is structured exactly like this package was before.

**Review by sanderr — APPROVED** [review]

**Pull request merged**
